src/yaml/from_yaml.cr

```diff
@@ -161,7 +181,12 @@ def Union.new(pull : YAML::PullParser)
 end
 
 def Time.new(pull : YAML::PullParser)
-  Time::Format::ISO_8601_DATE_TIME.parse(pull.read_scalar)
+  location = pull.location
+  begin
+    Time::Format::ISO_8601_DATE_TIME.parse(pull.read_scalar)
```

Comment:
[YAML timestamp type](http://yaml.org/type/timestamp.html) uses a specific subset of ISO 8601 which requires a custom parser. This doesn't need to be implemented in this PR but I'd recommend putting in a TODO note.